### PR TITLE
Update MONSTER_SPECIAL_ATTACKS.md 

### PR DESCRIPTION
**Pull request by RedMisao**
#### Summary
None

#### Purpose of change
1. Removed the obsolete `burst_limit` field from gun-type special attacks (there is no code for burst_limit and trying to use it gives an error)
2. Further described how `ammo_type` from gun-type special attacks works
3. Checked grammar and style, added and edited some other descriptions
4. Checked and added links
5. Moved some things to their own sections

### CODE REVIEW

**Comment by MNG-cataclysm** [comment]
“Documentation” isn’t a valid changelog ID. I’d replace it with something like “Infastructure”, or replace the entire line with “None”.

File: doc/MONSTER_SPECIAL_ATTACKS.md [hunk 1]
```diff
@@ -1,50 +1,69 @@
...
-- Descriptions for `Hardcoded special attacks` could stand to be more descriptive of exactly what the attack does.
+- Descriptions for [Hardcoded special attacks](#hardcoded-special-attacks) could stand to be more descriptive of exactly what the attack does.
+
 
-## Adding "special_attacks" to monsters
-(array of special attack definitions, optional)
+## Adding `special_attacks` to monsters
 
-Monster's special attacks. This should be an array, each element of it should be an object (new style) or an array (old style).
+Monster special attacks can be defined either as old style arrays, new style objects, or a combination of both.
```
Comment (Fris0uman) [inline doc/MONSTER_SPECIAL_ATTACKS.md:30]: Might want to add note that using old style should be avoided, I guess we probably want to phase it out eventually.

File: doc/MONSTER_SPECIAL_ATTACKS.md [hunk 2]
```diff
@@ -55,218 +74,222 @@ In the case of separately defined attacks the object has to contain at least an
...
+- ```BREATHE``` Spawns a `breather`.  Note: `breather hub` only!
 - ```CALLBLOBS``` Calls 2/3 of nearby blobs to defend this monster, and sends 1/3 of nearby blobs after the player.
-- ```CHICKENBOT``` LEGACY - Robot can attack with tazer, M4, or MGL depending on distance.
+- ```CHICKENBOT``` Robot can attack with tazer, M4, or MGL depending on distance.  Note: Legacy special attack.
 - ```COPBOT``` Cop-bot warns then tazes the player.
 - ```DANCE``` Monster dances.
 - ```DARKMAN``` Can cause darkness and wraiths to spawn.
-- ```DERMATIK_GROWTH``` Dermatik larva grows into an adult - obsoleted by them using `age_grow`.
 - ```DERMATIK``` Attempts to lay dermatik eggs in the player.
+- ```DERMATIK_GROWTH``` Dermatik larva grows into an adult.  Note: obsoleted by them using `age_grow`.
```
Comment (Fris0uman) [inline doc/MONSTER_SPECIAL_ATTACKS.md:103]: Maybe obsoleted function should not appear in the documentation

Comment (RedMisao) [reply doc/MONSTER_SPECIAL_ATTACKS.md:103]: My bad, seems I missed or forgot that's not used anymore. Removing it.

File: doc/MONSTER_SPECIAL_ATTACKS.md (hunk 1 above)
Comment (RedMisao) [reply doc/MONSTER_SPECIAL_ATTACKS.md:30]: Adding a suggestion in the doc.

**Pull request merged**